### PR TITLE
feat(cli): tree option in output flag for app sync, app wait and app rollback for tree view

### PR DESCRIPTION
**Pull request by schakrad**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: cmd/argocd/commands/app.go [hunk 1]
```diff
@@ -1539,6 +1540,7 @@ func NewApplicationWaitCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
+			output, _ = c.Flags().GetString("output")
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/app.go:1543]: I don't think this is needed. The `&output` reference below will make sure the `output` value is updated correctly.

File: cmd/argocd/commands/app.go [hunk 2]
```diff
@@ -1650,17 +1654,14 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
...
-
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
-
 			if len(args) > 1 && selector != "" {
 				log.Fatal("Cannot use selector option when application name(s) passed as argument(s)")
 			}
-
+			output, _ = c.Flags().GetString("output")
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/app.go:1664]: Ditto, not needed.

File: cmd/argocd/commands/app.go [hunk 3]
```diff
@@ -2341,17 +2371,18 @@ func NewApplicationRollbackCommand(clientOpts *argocdclient.ClientOptions) *cobr
...
 		Use:   "rollback APPNAME [ID]",
 		Short: "Rollback application to a previous deployed version by History ID, omitted will Rollback to the previous version",
 		Run: func(c *cobra.Command, args []string) {
 			ctx := c.Context()
-
 			if len(args) == 0 {
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
+			output, _ = c.Flags().GetString("output")
```
Comment (crenshaw-dev) [inline cmd/argocd/commands/app.go:2385]: Not needed

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Small things :-)

File: cmd/argocd/commands/app.go (hunk 3 above)
Comment (schakrad) [reply cmd/argocd/commands/app.go]: Removed the stmnt.

File: cmd/argocd/commands/app.go (hunk 2 above)
Comment (schakrad) [reply cmd/argocd/commands/app.go]: Made the change

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: cmd/argocd/commands/app.go [hunk 4]
```diff
@@ -2122,17 +2120,54 @@ func waitOnApplicationStatus(ctx context.Context, acdClient argocdclient.Client,
...
+				printAppResources(w, app)
+				_ = w.Flush()
+			}
+		case "tree":
+			_, appIf := acdClient.NewApplicationClientOrDie()
+			mapUidToNode, mapParentToChild, parentNode := parentChildDetails(appIf, ctx, appName, appNs)
+			mapNodeNameToResourceState := make(map[string]*resourceState)
+			for _, res := range getResourceStates(app, nil) {
+				mapNodeNameToResourceState[res.Kind+"/"+res.Name] = res
+			}
```
[the remark this answers is not in the document]
Comment (schakrad) [reply cmd/argocd/commands/app.go:2140]: Made the change. Thanks Michael.

File: cmd/argocd/commands/app.go [hunk 5]
```diff
@@ -2122,17 +2120,54 @@ func waitOnApplicationStatus(ctx context.Context, acdClient argocdclient.Client,
 			printOperationResult(app.Status.OperationState)
 		}
 
-		if len(app.Status.Resources) > 0 {
-			fmt.Println()
-			w := tabwriter.NewWriter(os.Stdout, 5, 0, 2, ' ', 0)
-			printAppResources(w, app)
-			_ = w.Flush()
+		switch output {
```
[the remark this answers is not in the document]
Comment (schakrad) [reply cmd/argocd/commands/app.go:2123]: Nops Michael,  It is not entirely duplicated.printHeader is a functionality that is extra in app get command, which runs for wide,tree and tree=detailed and not for json,yaml. I initially thought of refactoring that command , but printHeader function is preventing this.

**Comment by crenshaw-dev** [comment]
> Can we make the initial printout respect the -o flag? Looks like right now it always prints the table.

Ignore that one. I misunderstood how that initial printout is working. Let's leave it as a "table" view.

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**